metadata-service/openapi-servlet/src/main/java/io/datahubproject/openapi/operations/elastic/OperationsController.java

```diff
@@ -130,4 +146,86 @@ public ResponseEntity<String> getIndexSizes() {
             .collect(Collectors.toList()));
     return ResponseEntity.ok(j.toString());
   }
+
+  @Tag(name = "ElasticSearchOperations")
+  @GetMapping(path = "/explainSearchQuery", produces = MediaType.APPLICATION_JSON_VALUE)
+  @Operation(summary = "Explain Search Query")
+  public ResponseEntity<ExplainResponse> explainSearchQuery(
+      @Parameter(
+              name = "query",
+              required = true,
+              description =
+                  "Query to evaluate for specified document, will be applied as an input string to standard search query builder.")
+          @RequestParam("query")
+          @Nonnull
+          String query,
+      @Parameter(
+              name = "documentId",
+              required = true,
+              description = "Document ID to apply explain to.")
+          @RequestParam("documentId")
+          @Nonnull
+          String documentId,
+      @Parameter(
+              name = "entityName",
+              required = true,
+              description = "Name of the entity the document belongs to.")
+          @RequestParam("entityName")
+          @Nonnull
+          String entityName,
+      @Parameter(name = "from", required = true, description = "Start point for pagination.")
+          @RequestParam("from")
+          int from,
+      @Parameter(name = "size", required = true, description = "Page size for pagination.")
+          @RequestParam("size")
+          int size,
+      @Parameter(
+              name = "filters",
+              required = false,
+              description = "Additional filters to apply to query.")
+          @RequestParam("filters")
+          @Nullable
+          Filter filters,
+      @Parameter(
+              name = "sortCriterion",
+              required = false,
+              description = "Criterion to sort results on.")
+          @RequestParam("sortCriterion")
+          @Nullable
+          SortCriterion sortCriterion,
+      @Parameter(
+              name = "searchFlags",
+              required = false,
+              description = "Optional configuration flags.")
+          @RequestParam("searchFlags")
+          @Nullable
+          SearchFlags searchFlags,
+      @Parameter(
+              name = "facets",
+              required = false,
+              description = "List of facet fields for aggregations.")
+          @RequestParam("facets")
+          @Nullable
+          List<String> facets) {
+
+    Authentication authentication = AuthenticationContext.getAuthentication();
+    String actorUrnStr = authentication.getActor().toUrnStr();
+    DisjunctivePrivilegeGroup orGroup =
+        new DisjunctivePrivilegeGroup(
+            ImmutableList.of(
+                new ConjunctivePrivilegeGroup(
+                    ImmutableList.of(
+                        PoliciesConfig.GET_TIMESERIES_INDEX_SIZES_PRIVILEGE.getType()))));
```

Comment:
Timeseries index privilege probably doesn't make sense.

Reply:
ES_EXPLAIN_QUERY_PRIVILEGE

Reply:
Thanks for the catch, thought I changed this